src/bindings/haskell/README.md

```diff
@@ -4,12 +4,33 @@ A full haskell binding using c2hs.
 ## Usage
 The bindings are built with cabal, Haskell's default build system. It needs the 
 GHC, cabal and C2HS, which generates the actual bindings out of a metacode and 
-will be called during the build automatically (Call `cabal install c2hs` to 
-install it globally).
+will be called during the build automatically.
+
+There are some examples how the bindings can be used in
+[examples](src/bindings/haskell/examples). Furthermore there exists the possibility 
+to write [plugins](src/plugins/haskell/) using the haskell bindings.
+
+To build and test it manually call `cabal test`. Otherwise the bindings will be 
+installed along with elektra if the bindings are included in the build.
 
-To build and test it call `cabal test`.
 To open the bindings in ghci to play around with them call
-`cabal repl --ghc-options="-lelektra"`. 
+`cabal repl --ghc-options="-lelektra"` in this folder. 
+
+## Dependencies
+
+- GHC (Glasgow Haskell Compiler) > 8.0.0 (older versions may work but are untested)
+- cabal > 1.24.0.2  (older versions may work but untested, 2.0.0.0 works as well)
+
+GHC and cabal can be either obtained using a package manager or downloaded directly
+from [the Haskell homepage](https://www.haskell.org/platform/).
+
+Before installing the remaining dependencies, make sure to update your cabal database
+which contains the current list of haskell packages by invoking `cabal update`, this
+is not done automatically.
+
+- c2hs `cabal install c2hs` > 0.28.2 (older versions untested)
+- hspec `cabal install hspec` (only if the tests are being built, v2.4.4 tested)
+- QuickCheck `cabal install QuickCheck` (only if the tests are being built, v2.10.1 tested)
```

Comment:
For convenience you could give a single command which does everything.

Isn't it better to install c2hs from the package manager? (At least it is included in Debian.)

Reply:
if it is available in the package manager it most likely is. In homebrew its not available so therefore i am used to install it via cabal, but i'll add this recommendation